src/core/file.cc

```diff
@@ -1109,7 +1109,12 @@ make_file_impl(int fd, file_open_options options, int flags, struct stat st) noe
         internal::fs_info fsi;
         fsi.block_size = write_block_size;
         fsi.nowait_works = blockdev_nowait_works(st.st_dev);
-        fsi.align = internal::alignments { memory_dma_alignment, read_block_size, write_block_size, write_block_size };
+        fsi.align = internal::alignments {
+            static_cast<unsigned>(memory_dma_alignment),
+            static_cast<unsigned>(read_block_size),
+            static_cast<unsigned>(write_block_size),
+            static_cast<unsigned>(write_block_size)
+        };
```

Comment:
Does this change belong to the first patch in this PR?

Reply:
yes, should have been fixed.